### PR TITLE
Move from Dep To Modules

### PR DESCRIPTION
**Pull request by edwardcallahan**
Following this [SO](https://stackoverflow.com/questions/55664630/how-do-i-migrate-from-dep-to-go-modules) post.
`go test -race ./...` still passes.
`go test all` looks ok with the likes of
```
../../operator-framework/operator-sdk/pkg/ansible/controller/controller.go:35:2: cannot find package "sigs.k8s.io/controller-runtime/pkg/source" in any of:
	/usr/local/go/src/sigs.k8s.io/controller-runtime/pkg/source (from $GOROOT)
	/home/edc/go/src/sigs.k8s.io/controller-runtime/pkg/source (from $GOPATH)
../../operator-framework/operator-sdk/internal/pkg/scaffold/crd.go:31:2: cannot find package "sigs.k8s.io/controller-tools/pkg/crd/generator" in any of:
	/usr/local/go/src/sigs.k8s.io/controller-tools/pkg/crd/generator (from $GOROOT)
	/home/edc/go/src/sigs.k8s.io/controller-tools/pkg/crd/generator (from $GOPATH)
```

### CODE REVIEW

**Review by marcoderama — COMMENTED** [review]
I read every single line of this diff and it...
LGTM!

**Comment by edwardcallahan** [comment]
`go test all ` attached
[ACO-test-all.txt](https://github.com/lightbend/akka-cluster-operator/files/3719894/ACO-test-all.txt)

**Comment by edwardcallahan** [comment]
Test all
=========================
Ran 16 of 0 Specs in 0.002 seconds
FAIL! -- 0 Passed | 16 Failed | 0 Pending | 0 Skipped
--- FAIL: TestSource (0.00s)
FAIL
FAIL	sigs.k8s.io/controller-runtime/pkg/source	0.018s
ok  	sigs.k8s.io/controller-runtime/pkg/source/internal	(cached)
ok  	sigs.k8s.io/controller-runtime/pkg/webhook/admission	(cached)
?   	sigs.k8s.io/controller-runtime/pkg/webhook/admission/types	[no test files]
?   	sigs.k8s.io/controller-runtime/pkg/webhook/internal/metrics	[no test files]
?   	sigs.k8s.io/controller-runtime/pkg/webhook/types	[no test files]
ok  	sigs.k8s.io/testing_frameworks/integration	(cached)
ok  	sigs.k8s.io/testing_frameworks/integration/addr	(cached)
ok  	sigs.k8s.io/testing_frameworks/integration/internal	(cached)
ok  	sigs.k8s.io/yaml	(cached)
ok  	sort	(cached)
ok  	strconv	(cached)
ok  	strings	(cached)
ok  	sync	(cached)
ok  	sync/atomic	(cached)
ok  	syscall	(cached)
ok  	testing	(cached)
?   	testing/iotest	[no test files]
ok  	testing/quick	(cached)
ok  	text/tabwriter	(cached)
ok  	text/template	(cached)
ok  	text/template/parse	(cached)
ok  	time	(cached)
ok  	unicode	(cached)
ok  	unicode/utf16	(cached)
ok  	unicode/utf8	(cached)
?   	unsafe	[no test files]
?   	vendor/golang.org/x/crypto/chacha20poly1305	[no test files]
?   	vendor/golang.org/x/crypto/cryptobyte	[no test files]
?   	vendor/golang.org/x/crypto/cryptobyte/asn1	[no test files]
?   	vendor/golang.org/x/crypto/curve25519	[no test files]
?   	vendor/golang.org/x/crypto/hkdf	[no test files]
?   	vendor/golang.org/x/crypto/internal/chacha20	[no test files]
?   	vendor/golang.org/x/crypto/internal/subtle	[no test files]
?   	vendor/golang.org/x/crypto/poly1305	[no test files]
?   	vendor/golang.org/x/net/dns/dnsmessage	[no test files]
?   	vendor/golang.org/x/net/http/httpguts	[no test files]
?   	vendor/golang.org/x/net/http/httpproxy	[no test files]
?   	vendor/golang.org/x/net/http2/hpack	[no test files]
?   	vendor/golang.org/x/net/idna	[no test files]
?   	vendor/golang.org/x/net/nettest	[no test files]
?   	vendor/golang.org/x/sys/cpu	[no test files]
?   	vendor/golang.org/x/text/secure/bidirule	[no test files]
?   	vendor/golang.org/x/text/transform	[no test files]
?   	vendor/golang.org/x/text/unicode/bidi	[no test files]
?   	vendor/golang.org/x/text/unicode/norm	[no test files]

**Comment by edwardcallahan** [comment]
From test all:

Failure [0.002 seconds]
[BeforeSuite] BeforeSuite 
/home/edc/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/source/source_suite_test.go:42

  Unexpected error:
      <*errors.errorString | 0xc00020aa70>: {
          s: "failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory",
      }
      failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory
  occurred


Noted: https://github.com/deislabs/smi-adapter-istio/pull/27#pullrequestreview-257985192

**Comment by edwardcallahan** [comment]
kubebuilder dependencies do not appear to be specific to this 'test all', ie from [smi-adapter-istio # 27](https://github.com/deislabs/smi-adapter-istio/pull/27).

**Comment by edwardcallahan** [comment]
Repeated migration. Much more manageable results. 
Repeated on a fresh branch, results are same are `afc0444`. 

@marcoderama Please re-review. I cannot currently explain what changed, but this is makes more sense.

**Review by marcoderama — APPROVED** [review]

**Comment by edwardcallahan** [comment]
This requires a version increment. This is no longer 0.0.1!

File: deploy/olm-catalog/akka-cluster-operator/0.2.0/akka-cluster-operator.v0.2.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    - name: Intro to Akka
+      url: https://doc.akka.io/docs/akka/current/guide/introduction.html
+    - name: Intro to Akka Cluster
+      url: https://doc.akka.io/docs/akka/current/common/cluster.html
+    - name: Akka Cluster demo application
+      url: https://github.com/lightbend/akka-java-cluster-openshift
+    - name: Deploying a Lagom application to OpenShift
+      url: https://developer.lightbend.com/guides/openshift-deployment/lagom/index.html
+  version: 0.2.0
+  replaces: akka-cluster-operator.v0.0.1
```
Comment (marcoderama) [inline deploy/olm-catalog/akka-cluster-operator/0.2.0/akka-cluster-operator.v0.2.0.clusterserviceversion.yaml:300]: Fwiw, I never had good luck specifying a `replaces` value.  I ended up being hesitant to use it...

File: hacking.md [hunk 2]
```diff
@@ -26,6 +26,12 @@ code. Since this area is a fast moving target, the main thought here was just to
 native client libs which happen to be in go, so this is in go too. But this could all be
 written in Scala if one wanted to be adventurous.
 
+## Developer Setup
+
+Use of [Go modules](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#go-modules),
+per the [Go modules Wiki](https://github.com/golang/go/wiki/Modules), requires invoking go outside of the `src` folder
```
Comment (marcoderama) [inline hacking.md]: Should 'go' in "requires invoking go outside of..." be capitalized?

**Review by marcoderama — APPROVED** [review]
With some very small comments.

**Comment by edwardcallahan** [comment]
```
(openshift) edc@AJGoddard ~/go/src/github.com/lightbend/akka-cluster-operator $ operator-sdk version
operator-sdk version: "v0.11.0", commit: "39c65c36159a9c249e5f3c178205cc6e86c16f8d", go version: "go1.12.7 linux/amd64"


(openshift) edc@AJGoddard ~/go/src/github.com/lightbend/akka-cluster-operator $ operator-sdk scorecard
{
  "metadata": {
    "creationTimestamp": null
  },
  "log": "time=\"2019-10-21T15:49:30-07:00\" level=info msg=\"Using config file: /home/edc/go/src/github.com/lightbend/akka-cluster-operator/.osdk-scorecard.yaml\"\n",
  "results": [
    {
      "name": "Basic Tests",
      "description": "Test suite that runs basic, functional operator tests",
      "error": 0,
      "pass": 3,
      "partialPass": 0,
      "fail": 0,
      "totalTests": 3,
      "totalScorePercent": 100,
      "tests": [
        {
          "state": "pass",
          "name": "Spec Block Exists",
          "description": "Custom Resource has a Spec Block",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Status Block Exists",
          "description": "Custom Resource has a Status Block",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Writing into CRs has an effect",
          "description": "A CR sends PUT/POST requests to the API server to modify resources in response to spec block changes",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        }
      ],
      "log": "time=\"2019-10-21T15:49:30-07:00\" level=info msg=\"Running for cr: deploy/crds/app_v1alpha1_akkacluster_cr.yaml\"\n\n---\n"
    },
    {
      "name": "OLM Tests",
      "description": "Test suite checks if an operator's CSV follows best practices",
      "error": 0,
      "pass": 5,
      "partialPass": 0,
      "fail": 0,
      "totalTests": 5,
      "totalScorePercent": 100,
      "tests": [
        {
          "state": "pass",
          "name": "Provided APIs have validation",
          "description": "All CRDs have an OpenAPI validation subsection",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Owned CRDs have resources listed",
          "description": "All Owned CRDs contain a resources subsection",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [
            "If it would be helpful to an end-user to understand or troubleshoot your CR, consider adding resources [configmaps/v1] to the resources section for owned CRD AkkaCluster"
          ],
          "errors": []
        },
        {
          "state": "pass",
          "name": "CRs have at least 1 example",
          "description": "The CSV's metadata contains an alm-examples section",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Spec fields with descriptors",
          "description": "All spec fields have matching descriptors in the CSV",
          "earnedPoints": 2,
          "maximumPoints": 2,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Status fields with descriptors",
          "description": "All status fields have matching descriptors in the CSV",
          "earnedPoints": 4,
          "maximumPoints": 4,
          "suggestions": [],
          "errors": []
        }
      ],
      "log": "time=\"2019-10-21T15:49:49-07:00\" level=info msg=\"Running for cr: deploy/crds/app_v1alpha1_akkacluster_cr.yaml\"\n\n---\n"
    }
  ]
}
```

**Pull request merged**